modules/userId.js

```diff
@@ -122,6 +122,73 @@ export const pubCommonIdSubmodule = {
   }
 };
 
+// @type {Submodule}
+export const digitrustIdModule = {
+  name: 'digitrust',
+  decode(value) {
+    let decodedId;
+    try {
+      // decode the id per DigiTrust lib
+      decodedId = atob(value);
+      return {
+        'digitrustid': decodedId
+      }
+    } catch (e) {
+      return undefined;
+    }
+  },
+  getId() {
+    const MAX_RETRIES = 4;
+    let retries = 0;
+
+    return function (callback) {
+      function initDigiTrust() {
+        if (((typeof window.DigiTrust !== 'undefined' && window.DigiTrust == null) ? false : DigiTrust.isClient) === false) {
+          if (retries < MAX_RETRIES) {
+            // set timeout with extended time to check for the DigiTrust frame work
+            setTimeout(initDigiTrust, 100 * (1 + retries++));
```

Comment:
probably can get rid of the RETRY logic since this won't run until after the auction + any sync delay, which should give the DT library plenty of time to load.  Although this isn't much code after being minified.